### PR TITLE
Replace occurences of keySet() with entrySet() when value are needed

### LINKED ISSUE #634: Replace occurences of keySet() with entrySet() when value are needed
### Description
There are few places in our codebase where keySet() is used instead of entrySet(), which leads to a sonar code smell.

#### "entrySet()" should be iterated when both the key and value are needed.

We need to replace these occureneces of keySet() with entrySet() to make it SonarCloud compliant![0]

[0] https://sonarcloud.io/organizations/jkubeio/rules?open=java%3AS2864&rule_key=java%3AS2864

### PR DESCRIPTION
**Pull request by theexplorist**
## Description
Fixes #634 

Replace occurences of keySet() with entrySet() when value are needed

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractor.java [hunk 1]
```diff
@@ -18,184 +18,192 @@
...
+
+	/**
+	 * Flattens a nested map into a Map<String, String>.
+	 * 
+	 * @param map The target map.
+	 * @return The flattened map.
+	 */
+	private Map<String, String> flatten(Map map) {
+		Map<String, String> flat = new HashMap<>();
+		Iterator mapEntries = map.entrySet().iterator();
```
Comment (theexplorist) [inline jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractor.java:150]: Getting Iterator for raw type map

**Comment by theexplorist** [comment]
> SonarCloud Quality Gate failed.
> 
> [Bug](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=BUG)
> [Vulnerability](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=VULNERABILITY)
> [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=jkubeio_jkube&pullRequest=646&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=jkubeio_jkube&pullRequest=646&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=jkubeio_jkube&pullRequest=646&resolved=false&types=SECURITY_HOTSPOT)
> [Code Smell](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=CODE_SMELL) [2 Code Smells](https://sonarcloud.io/project/issues?id=jkubeio_jkube&pullRequest=646&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=jkubeio_jkube&pullRequest=646&metric=new_coverage&view=list) [41.7% Coverage](https://sonarcloud.io/component_measures?id=jkubeio_jkube&pullRequest=646&metric=new_coverage&view=list)
> [](https://sonarcloud.io/component_measures?id=jkubeio_jkube&pullRequest=646&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=jkubeio_jkube&pullRequest=646&metric=new_duplicated_lines_density&view=list)

@manusa @rohanKanojia there are two code smells regarding parametrisation but the map itself is raw type!

**Comment by rohanKanojia** [comment]
I think it's okay. We can even proceed without fixing this(I don't think we can change this).

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx

**Pull request merged**